### PR TITLE
#36553 specialisation error 520 is misleading

### PR DESCRIPTION
**Pull request by giannicic**
this commit corrects E0520 error text.
See referenced issue for further info

r? @nrc

### CODE REVIEW

File: src/librustc_typeck/check/mod.rs [hunk 1]
```diff
@@ -904,15 +904,15 @@ fn report_forbidden_specialization<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
...
         impl_item.name);
     err.span_label(impl_item.span, &format!("cannot specialize default item `{}`",
                                             impl_item.name));
 
     match tcx.span_of_impl(parent_impl) {
         Ok(span) => {
             err.span_label(span, &"parent `impl` is here");
-            err.note(&format!("to specialize, either the parent `impl` or `{}` \
+            err.note(&format!("to specialize, `{}` \
                                in the parent `impl` must be marked `default`",
```
Comment (nrc) [inline src/librustc_typeck/check/mod.rs]: nit: this string could all fit on one line now.

**Comment by nrc** [comment]
looks good, thanks! r=me with the nit about the string fixed.

**Comment by giannicic** [comment]
r=nrc

**Pull request merged**